src/components/tire/AddTireCheckPage.js

```diff
@@ -0,0 +1,257 @@
+import { useEffect, useState } from "react";
+import Breadcrumb, { BreadcrumbIndicies } from "../breadcrumb/Breadcrumb";
+
+import styles from "./AddTireCheckPage.module.css";
+import sharedStyles from "../SharedCard.module.css";
+import sharedButtons from "../SharedButton.module.css";
+import NumberIncrementer from "../mileage/NumberIncrementer";
+
+import {
+  getActiveCar,
+  getLastTireCheckEvent,
+  addTireCheckEvent,
+} from "../DBHelperFunctions";
+import dayjs from "dayjs";
+
+export default function AddTireCheckPage() {
+  const [currentTirePressure, setCurrentTirePressure] = useState({
+    topLeft: 0,
+    topRight: 0,
+    bottomLeft: 0,
+    bottomRight: 0,
+  });
+  const [initialTirePressure, setInitialTirePressure] =
+    useState(currentTirePressure);
+  const [globalTirePressure, setGlobalTirePressure] = useState(30);
+  const [initialGlobalTirePressure, setInitialGlobalTirePressure] =
+    useState(globalTirePressure);
+  const [activeCar, setActiveCar] = useState(undefined);
+
+  useEffect(() => {
+    getActiveCar((car) => {
+      const currentTime = dayjs();
+      setActiveCar(car);
+      getLastTireCheckEvent(car["id"], currentTime.valueOf()).then((events) => {
+        if (events.length !== 0) {
+          const currentPressure = events[0]["pressure"];
+          setGlobalTirePressure(currentPressure);
+          setInitialGlobalTirePressure(currentPressure);
+          const initialPressure = {
+            topLeft: currentPressure,
+            topRight: currentPressure,
+            bottomLeft: currentPressure,
+            bottomRight: currentPressure,
+          };
+          console.debug({ initialPressure, events });
+          setInitialTirePressure(initialPressure);
+          setCurrentTirePressure(initialPressure);
+        }
+      });
+    });
+  }, []);
+
+  const setCurrentTire = (value, key) => {
+    const newTirePressure = { ...currentTirePressure, [key]: value };
+    console.log({
+      currentTirePressure,
+      newTirePressure,
+    });
+    setCurrentTirePressure(newTirePressure);
+  };
+
+  const handleSave = () => {
+    // save to db
+    // then set current tire pressure and initial tire pressure to global
+
+    addTireCheckEvent({
+      ...currentTirePressure,
+      pressure: globalTirePressure,
+      carId: activeCar["id"],
+    }).then((ids) => {
+      const newTirePressure = {
+        topLeft: globalTirePressure,
+        topRight: globalTirePressure,
+        bottomLeft: globalTirePressure,
+        bottomRight: globalTirePressure,
+      };
+      setCurrentTirePressure(newTirePressure);
+      setInitialTirePressure(newTirePressure);
+      setInitialGlobalTirePressure(globalTirePressure);
+    });
+  };
+
+  const resetTirePressure = () => {
+    setCurrentTirePressure(initialTirePressure);
+    setGlobalTirePressure(initialGlobalTirePressure);
+  };
+
+  const hasChanged = () => {
+    const hasChanged =
+      currentTirePressure !== initialTirePressure ||
```

Comment:
Take note that there is a slight bug here, where it will not do a deep assertion, it will just assert the objects themselves, meaning that if you went from tire pressure of 33 to 32 and then back to 33 it will still mark the page as having changed.

A fix would be to manually assert each of the tire pressures.